### PR TITLE
Add alignment syntax to parser

### PR DESCRIPTION
**Pull request by lubegasimon**
Open in favour of https://github.com/ocaml/odoc/pull/619#discussion_r587734198

### CODE REVIEW

File: src/parser/syntax.ml [hunk 1]
```diff
@@ -890,6 +890,24 @@ let rec block_element_list :
...
+        junk input;
+        let content, brace_location =
+          delimited_inline_element_list ~parent_markup:token
+            ~parent_markup_location:location ~requires_leading_whitespace:true
+            input
+        in
+        let location = Location.span [ location; brace_location ] in
+        if content = [] then
+          Parse_error.should_not_be_empty ~what:(Token.describe token) location
+          |> Error.warning input.warnings;
```
Comment (Julow) [inline src/parser/syntax.ml:903]: This should be removed.
Instead, there should be a warning saying that the markup won't be used and should be removed from the sources.

**Review by Julow — COMMENTED** [review]
One comment, otherwise looks good.

**Comment by jonludlam** [comment]
@lubegasimon there are conflicts in `Ocamlary/index.html` - could you please rebase then we can merge this.

**Comment by lubegasimon** [comment]
> @lubegasimon there are conflicts in `Ocamlary/index.html` - could you please rebase then we can merge this.

done

**Comment by jonludlam** [comment]
Thanks!

**Pull request merged**